src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArray = [];
+  const temp = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      const addKeys = action.extraData;
+
+      Object.assign(temp, addKeys);
+    }
```

Comment:
you can use object destructing for `action`
```suggestion
  for (const action of actions) {
    const { type, extraData, keysToRemove } = action;
    
    if (type === 'addProperties') {
      Object.assign(temp, extraData);
    }
```